File: src/services/x/auth/xAuthConstants.ts
export const X_BASE_URL = 'https://x.com';
export const X_LOGIN_URL = 'https://x.com/i/flow/login';

// Cookie names derived from Rettiwt-API's IAuthCookie interface (types/auth/AuthCookie.ts)
// and AuthCookie model (models/auth/AuthCookie.ts).
// Change: split into required vs optional — Rettiwt-API treats all four as mandatory,
// but here `kdt` is optional since it isn't always set during WebView login.
export const REQUIRED_X_COOKIES = ['auth_token', 'ct0', 'twid'] as const;
export const OPTIONAL_X_COOKIES = ['kdt'] as const;
export const COOKIE_ORDER = ['auth_token', 'ct0', 'kdt', 'twid'] as const;

export const POST_LOGIN_PATH_HINTS = [
  '/home',
  '/notifications',
  '/messages',
  '/explore',
  '/settings',
  '/compose',
  '/search',
  '/i/bookmarks',
] as const;

File: android/app/src/main/kotlin/com/yonisirote/readmyfeed/x/auth/XAuthConstants.kt
package com.yonisirote.readmyfeed.x.auth

const val X_BASE_URL = "https://x.com"
const val X_LOGIN_URL = "$X_BASE_URL/i/flow/login"

val X_ALLOWED_ORIGINS = listOf(
  X_BASE_URL,
  "https://twitter.com",
  "https://mobile.twitter.com",
)

val REQUIRED_X_COOKIES = listOf("auth_token", "ct0", "twid")
val OPTIONAL_X_COOKIES = listOf("kdt")
val COOKIE_ORDER = listOf("auth_token", "ct0", "kdt", "twid")

val POST_LOGIN_PATH_HINTS = listOf(
  "/home",
  "/notifications",
  "/messages",
  "/explore",
  "/settings",
  "/compose",
  "/search",
  "/i/bookmarks",
)

internal const val X_SESSION_PREFERENCES_NAME = "x_session_preferences"
internal const val X_SESSION_KEY = "x-auth-session"
